### ISSUE TITLE
Biocgenerics is being imported for a single line and it pulls a huge loads of dependencies that make it difficult to install in conda

### ISSUE THREAD

**Issue by pcm32** [ebi-gene-expression-group/internal-ExpressionAtlas#2]
Biocgenerics seems to only be used here:

https://github.com/ebi-gene-expression-group/internal-ExpressionAtlas/blob/master/R/functions.R#L692

Yet it drags so many dependencies. We should try to avoid that call and remove it as a dependency.

**Comment by pinin4fjords** [comment]
This is a core package required for many Bioconductor packages, no? e.g. summarizedExperiment a couple of lines later likely wouldn't work without it.

**Comment by pcm32** [comment]
Yes, but then that package will request it upon install if it needs it, correct?

**Comment by pinin4fjords** [comment]
I don't think removing it as a dependency will make any difference- it will get pulled as an upstream dependency of everything else.

**Comment by pcm32** [comment]
My point, and what I see often in conda, is that dependency failures arise when multiple packages request the same dependency. This is specially problematic in R, where certain packages will be bound due to lack of updates to older versions of R. This is compounded when in turn you have too many dependencies.

**Comment by pinin4fjords** [comment]
Agreed on the tidy up, it's probably not required in the list of dependencies. 

But as far as the issue your report above goes, it won't make a difference to the number of packages that get installed (the 'so many dependencies' issue above), BiocGenerics will be installed anyway, and there's no need to remove the call to the package method.
